@here/harp-mapview/lib/text/TextElementsRenderer.ts

```diff
@@ -1709,8 +1727,14 @@ export class TextElementsRenderer {
         }
         // ... and render the icon (if any).
         if (iconReady) {
-            if (iconRejected) {
-                iconRenderState!.startFadeOut(renderParams.time);
+            if (iconRejected || iconInvisible) {
+                if (iconRenderState.isReallyVisible()) {
+                    iconRenderState.startFadeOut(renderParams.time);
+                } else if (!iconRenderState.isFadingOut()) {
+                    // Reset the icon and keep it from popping up before fading in.
+                    iconRenderState.reset();
+                    iconRenderState.opacity = 0;
```

Comment:
I don't understand the need of this.
If the label is not visible and not fading out then it must be in undefined or faded out state, either way you don't need to reset it.

Reply:
I think that is because we're not check for undefined in startFadeOut(). If the label is in undefined state, startFadeOut() should return early like for FadedOut and FadingOut states.

Reply:
Simplified back kto

            if (iconRejected) {
                iconRenderState.startFadeOut(renderParams.time);
            } else {
                iconRenderState!.startFadeIn(renderParams.time, this.m_options.disableFading);
            }

Reply:
... and defaulting the opacity value in RenderState to 0.0 instead of 1.0, which leads to issues.